### ISSUE TITLE
Single-quoted solitary item in for loop spuriously triggers SC2041 but not SC2043

### ISSUE THREAD

**Issue by antifuchs** [koalaman/shellcheck#2891]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2041
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

for dev in '/dev/disk/by-uuid/28DD-A89E'; do
  echo "$dev"
done
```

#### Here's what shellcheck currently says:
```
Line 3	SC2041: This is a literal string. To run as a command, use $(..) instead of '..' . 
```
#### Here's what I wanted or expected to see:

I would not expect this to warn about SC2041, but SC2043. The following snippet where a second single-quoted element is iterated over does not warn about SC2041 (and, correctly, not about SC2043):

```sh
#!/usr/bin/env bash

for dev in '/dev/disk/by-uuid/28DD-A89E' 'foo'; do
  echo "$dev"
done
```

and neither does the following with double quotes (it does warn about SC2043, as expected):

```sh
#!/usr/bin/env bash

for dev in "double-quoted"; do
  echo "$dev"
done
```

**Comment by ormaaj** [comment]
That word doesn't require quotes anyway so... maybe it's just guessing that it's a command-in-a-string for some reason. Erroneous if so.

[This](https://github.com/koalaman/shellcheck/wiki/SC2041) is just silly and hopefully not the actual rationale. There are several noisy and not super useful single-quoting warnings similar to this - where it tries to guess that the content of a single-quoted word was supposed to be something else.
